### PR TITLE
Scripts/Utgarde Pinnacle: Svala Sorrowgrave Intro and Ritual of the Sword

### LINKED ISSUE #23610: Boss Svala
**Description:**
Svala Sorrowgrave does not make it to the final part of the encounter.

CHANGEME Description of the problem or issue here.

**Current behaviour:**
After entry to the chamber where Svala is the Lich King appears as intended and turns her into Svala Sorrowgrave. Afterwards nothing happens but the 3 spawns show up that you have to kill during the encounter but she does not she says nothing back to the Lich King it ends there ans she remains unattackable.

CHANGEME Tell us what happens.
If this is a crash, post the crashlog (upload to https://gist.github.com/).

**Expected behaviour:**
Svala once transformed into Sorrowgrave should talk to the King and and then he asks her to destroy our uninvited guests and she in return says she will be happy to slaughter them. Then a second or so later she becomes attackable and the fight starts

CHANGEME Tell us what should happen instead.

**Steps to reproduce the problem:**
Enter Utgarde Pennicle clearing all the trash mobs making your way to the chamber where Svala stands. Once in far enough the initial encounter begins.

1. CHANGEME Step 1 include entries of affected creatures / items / quests with a link to the relevant wowhead page.  
2. Step 2
3. Step 3

**Branch(es):**
3.3.5 

CHANGEME 3.3.5, master or both

**TC rev. hash/commit:** 
374597c8e96c+

CHANGEME Copy the result of server debug command (if you need to run it from client get prat addon)

**Operating system:** CHANGEME OS
Windows 10 Pro

### PR DESCRIPTION
**Pull request by wotlk-enthusiast**
…Sword

**Changes proposed:**
- Added fix for event failure after UpdateEntry for Svala Sorrowgrave 
- Added flags and corrected timing for Ritual of the Sword
- Added auto-start combat with nearest target after event_intro is completed.

**Target branch(es):** 3.3.5

- [X] 3.3.5
- [ ] master

**Issues addressed:** Closes #23610 

**Tests performed:** (Does it build, tested in-game, etc.)
- [X] Builds in-game
- [X] Scripted intro event works with one player
- [X] Ritual of the sword works with one player

**Known issues and TODO list:** (add/remove lines as needed)
- Svala is supposed to be flying while she is being transformed (cosmetic issue)

### CODE REVIEW

**Comment by wotlk-enthusiast** [comment]
Feedback on the propriety of my pull request kindly requested. :smile:

**Comment by illfated** [comment]
PR title suggestion (based on conservative style): 
- **Scripts/Utgarde Keep: Svala Sorrowgrave intro & Ritual of the Sword**

**Comment by wotlk-enthusiast** [comment]
Utgarde Keep or Utgarde Pinnacle? Script is in Pinnacle but the folder is named UtgardeKeep.

**Comment by illfated** [comment]
```cpp
/home/circleci/project/src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp:349:41:
 fatal error: cannot initialize a parameter of type 'Unit *' with an rvalue of type 'Player *'
                            AttackStart(me->SelectNearestPlayer(100.0f));
                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

/home/circleci/project/src/server/game/AI/ScriptedAI/ScriptedCreature.h:157:42:
 note: passing argument to parameter here
        void AttackStart(Unit* /*target*/) override;
                                         ^
```

**Comment by illfated** [comment]
Sorry about that, I misread the script folder path and overlooked Pinnacle.
- https://wow.gamepedia.com/Svala_Sorrowgrave >>> Utgarde Pinnacle

Corrected suggestion: **Scripts/Utgarde Pinnacle: Svala Sorrowgrave intro & Ritual of the Sword**

**Comment by wotlk-enthusiast** [comment]
Yep, I saw that.  I'm recompiling locally at the moment with the following fix (before I make it a commit): 
```c++
AttackStart(me->SelectNearestPlayer(100.0f)->ToUnit());
```

**Comment by wotlk-enthusiast** [comment]
From a quick code review, looks like the proper fix to this is to do something like
```c
{some unit}->AI()->AttackStart(target)
```

**Comment by illfated** [comment]
Sure, you need to redo the `AttackStart()` to enable the extra argument to the call:
```cpp
void AttackStart(Unit* *target*) override;
```
Some form of this method may be needed in the init part of the script.

**Comment by jackpoz** [comment]
> ```c++
> cannot initialize a parameter of type 'Unit *' with an rvalue of type 'Player *'
> ```

This usually means the compiler doesn't know that Player inherits from Unit, for example if Player was forward declared.

Doing 
```cpp
#include "Player.h"
```

should fix it

**Comment by illfated** [comment]
Right. Of course. I should have remembered. That is what what the **nopch** part is about in

**ci/circleci: nopch**

**Comment by wotlk-enthusiast** [comment]
Sorry for branch/merge, I was trying to get rid of my old school account on the commit log, but oh well :shrug:

File: src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp [hunk 1]
```diff
@@ -364,6 +368,11 @@ class boss_svala : public CreatureScript
                                 DoCast(sacrificeTarget, SPELL_RITUAL_PREPARATION);
                                 SetCombatMovement(false);
                                 DoCast(me, SPELL_RITUAL_OF_THE_SWORD);
+                                me->SetControlled(true,UNIT_STATE_ROOT);
+                                me->SetDisableGravity(true);
+                                me->SetCanFly(true);
+                                me->SetHover(true);
```
Comment (Keader) [inline src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp:374]: looks wrong to me
never saw a creature with can fly, disable gravity and hover at same time.

File: src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp [hunk 2]
```diff
@@ -383,6 +393,18 @@ class boss_svala : public CreatureScript
...
+                            events.SetPhase(NORMAL);
+                            me->SetControlled(false, UNIT_STATE_ROOT);
+                            me->SetDisableGravity(false);
+                            me->SetCanFly(false);
+                            summons.DespawnAll();
+                            me->RemoveByteFlag(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, UNIT_BYTE1_FLAG_ALWAYS_STAND | UNIT_BYTE1_FLAG_HOVER);
+                            me->GetMotionMaster()->MoveFall();
+                            DoZoneInCombat();
+                            me->SetReactState(REACT_AGGRESSIVE);
+                            AttackStart(me->SelectNearestPlayer(100.0f));
```
Comment (Keader) [inline src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp:406]: what happens with cant find any player in 100 yards? will attack null?
will not crash?

File: src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp (hunk 1 above)
Comment (wotlk-enthusiast) [reply src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp:374]: SetHover was my addition to make her wings flap, the other two were in the previous commit that Killyana diff'd (https://github.com/TrinityCore/TrinityCore/issues/23610#issuecomment-529606023)

Thinking back, I think the thing that made her wings flap was not SetHover, but the flags from L375.  I'll remove and see if the visual is still there.

File: src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp (hunk 2 above)
Comment (wotlk-enthusiast) [reply src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp:406]: Would probably crash.  I'll add an if statement around it.

Comment (Keader) [reply src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp:406]: Try this:

```cpp
me->SetReactState(REACT_AGGRESSIVE);
DoZoneInCombat();
```
without AttackStart call, it should make she attack

File: src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp (hunk 1 above)
Comment (Keader) [reply src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp:374]: you checked with sniffs what should happens?

File: src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp (hunk 2 above)
Comment (wotlk-enthusiast) [reply src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp]: Resolved with 2a02067

File: src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp (hunk 1 above)
Comment (wotlk-enthusiast) [reply src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp]: Resolved with 2a02067

File: src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp (hunk 2 above)
Comment (wotlk-enthusiast) [reply src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp]: Didn't see your comment @Keader , sorry! Would that be better than what I did in 2a02067?

File: src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp (hunk 1 above)
Comment (wotlk-enthusiast) [reply src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp]: @Keader I was unaware that we had sniffs for this boss.  I certainly don't have access to them (and not sure exactly what to do with them if I had them :fearful: )

File: src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp (hunk 2 above)
Comment (jackpoz) [reply src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp]: Most AttackStart() implementations do a null check for the passed in parameter and if they don't they should as that's the de-facto standard. Just check UnitAI::AttackStart() .

**Comment by jackpoz** [comment]
is this ready to be merged ?

**Comment by illfated** [comment]
Looks like an unresolved question whether it is better to use `DoZoneInCombat();` instead of `AttackStart(target);` - from what I can see in the conversation with Keader.

**Comment by wotlk-enthusiast** [comment]
No, you're right @illfated. I think it's ready, unless I should change to the ```DoZoneInCombat();```.  I'm still unsure on the details, but I added the redundant error check.

**Comment by Riztazz** [comment]
I think there is a lot of code that is not required, i would go about it this way
https://gist.github.com/Riztazz/8517225a44d503ee4de77dc881e3458c

**Comment by Riztazz** [comment]
There is a lot of code in here that is not exactly necessary, if i recall correctly - most of the movement issues are related to lack of movement data in database, something along the lines of this:
https://gist.github.com/Riztazz/8517225a44d503ee4de77dc881e3458c

**Comment by Rushor** [comment]
or

```
diff --git a/src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp b/src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp
index f00ba0655f..0b6d59e7db 100644
--- a/src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp
+++ b/src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp
@@ -244,7 +244,7 @@ class boss_svala : public CreatureScript
 
                 events.Update(diff);
 
-                if (!_sacrificed && HealthBelowPct(50))
+                if (!_sacrificed && HealthBelowPct(50) && events.IsInPhase(NORMAL))
                 {
                     _sacrificed = true;
                     events.SetPhase(SACRIFICING);
```

like you sai in the other issue :P

**Comment by Riztazz** [comment]
Health related stuff is usually checked in DamageTaken hooks but yeah, that approach works as well

**Comment by illfated** [comment]
@Rushor : Cosmetic detail, use **```diff** at the beginning of your code block to highlight the code diff. 😉 👍

**Comment by wotlk-enthusiast** [comment]
Someone with access to sniffs want to check this against them and see what flags are actually necessary?

**Comment by Riztazz** [comment]
Is it really needed to be confirmed by sniffs tho? It hovers over the ground and flies at times - that alone tells us that this object required flying related stuff;p

**Comment by wotlk-enthusiast** [comment]
Been AFK a while, had to move.

I see Feedback-FixOutdatedMissingWIP label added, but I'm unsure what the consensus is on what I should fix. @Riztazz @Rushor @illfated Where do I go?

**Comment by illfated** [comment]
Thanks for asking, I don't have the exact answer. I suggest looking at the suggestions from Riztazz and Rushor to see which solution you think fits best with fixing the hover & fly movements.

**Comment by wotlk-enthusiast** [comment]
Alright, so I've been working on this for a bit now, and I've found two issues.

1. In https://github.com/TrinityCore/TrinityCore/blob/9683ce99ebef251f1c3c004d67bfbd26b49895c6/src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp#L274-L277
adding the code suggested by @Rushor and the DB updates suggested by @Riztazz still doesn't make her fly while she's transformed.  I don't know what I should try next. Maybe setting raw flags?  Why isn't she moving?  Is 
```cpp
me->SetDisableGravity(true,false);
me->GetMotionMaster()->MoveTakeoff(0, pos);
``` 
not sufficient to make her fly? What is the correct way to do this?

2. My code (and the diff suggested above by Rushor and Rizztazz) both have a flaw - after she does the sacrifice, if you kill the ritual summoners and just walk away from her (say, to the stairs), she will look at you, then regen health and reset.  Coming down from her flying point and running this:

```cpp
me->SetReactState(REACT_AGGRESSIVE);
DoZoneInCombat();
if (Unit* target = SelectTarget(SELECT_TARGET_MAXTHREAT,0,100.0f,true))
{
    AttackStart(target);
}
```

doesn't force her into combat.  You can reset the whole encounter by just walking away and not attacking her.  How should we force her to attack the person with the most threat (as the encounter is supposed to work)?

**Comment by illfated** [comment]
Good & fair question. I hope some of the coders will be able to help you further.
(For all I know, it could be a LoS problem, pathing problem or other.)

**Comment by wotlk-enthusiast** [comment]
OK, here's my current problem:

Setting her movement flags to DisableGravity doesn't seem to have an effect.  Here's what I did

```sql
DELETE FROM creature_template_movement WHERE CreatureId=29281;
INSERT INTO creature_template_movement values (29281,1,0,1,0,0,0);
```

and to verify:
```sql
SELECT * FROM creature_template_movement WHERE CreatureId=29281;
+------------+--------+------+--------+--------+-------+--------+
| CreatureId | Ground | Swim | Flight | Rooted | Chase | Random |
+------------+--------+------+--------+--------+-------+--------+
|      29281 |      1 |    0 |      1 |      0 |     0 |      0 |
+------------+--------+------+--------+--------+-------+--------+
```

where ```29281``` is her creatureID.

Then, I unbound myself from the instance, restarted the server, flushed my cache, unbound myself again to be sure, and went back to her, and ran .npc info and I get the following
```
NPC Name: Svala
Entry: 29281
Movement Type: Flight: None
```

What am I doing wrong here? How/Where do I set the flags so that they are reflected in-game? The simplest things seem to be throwing me for a loop......

**Comment by illfated** [comment]
@Killyana : Do you have any advice to offer? I feel a bit rusty on the questions above.

**Comment by Killyana** [comment]
Difficultyentry1=30809 for heroic

**Comment by illfated** [comment]
Thanks, @Killyana.

@wotlk-enthusiast : is it likely that you were targeting the wrong `creature_entry` / `CreatureId` for your testing?

**Comment by wotlk-enthusiast** [comment]
Yeah, that was the problem.  Switching to normal mode made it so that my flags were taking effect.  You'd think that .npc info would show a different entry if you were in heroic, but I guess that's not the case.  

Well, that fixes that.  But the new problem is that she always hovers about 3 yards off the ground and runs through the air to attack you.  I tried changing a bunch of flags, and cannot get her to either 1) come to the ground or 2) use her wings to fly around (like how youtube videos of the encounter show her moving).  Any idea on that one?

Also while I have your attention, is there some documentation or something that I should be reading instead of asking these seemingly simple questions?  Or are all new coders just doomed to ride the strugglebus and as a community we're just wasting everyone's time?  Gotta say, this is all so frustrating that I am highly discouraged from ever trying to fix anything like this again. :-1:

**Comment by Faq** [comment]
Well, when you figure how everything works and in the end she will do like master commands, then You can say that received some XP from all this :)

And here are more options to try out: https://github.com/TrinityCore/TrinityCore/issues/24656#issuecomment-630752811 and there linked PR what helped in that case, check if something can help here.

**Comment by illfated** [comment]
I wonder if you could find any useful information in the Val'kyrs in the Icecrown area, like the ones in the Ymirheim mines (Slaves to Saronite), or maybe even the Trial of the Champion val'kyr twins scripts.

**Comment by Killyana** [comment]
About the Boss movements you must use:
``` sql
DELETE FROM `creature_template_movement` WHERE `CreatureId` IN (29281,30809 );
INSERT INTO creature_template_movement (`CreatureId`, `Ground`, `Flights`) values
(29281,2,1),(30809 ,2,1);
``` 
Also during the transformation into a val'kyr you must add bytes1=33554432 to the boss to get the correct movements (wings and visual flying )

**Comment by wotlk-enthusiast** [comment]
> Well, when you figure how everything works and in the end she will do like master commands, then You can say that received some XP from all this :)

Are you sure you want to learn "TrinityCore Scripting" as your second skill? Allows for a maximum skill of 75...

> And here are more options to try out: [#24656 (comment)](https://github.com/TrinityCore/TrinityCore/issues/24656#issuecomment-630752811) and there linked PR what helped in that case, check if something can help here.

I'll check that out, thanks.


> I wonder if you could find any useful information in the Val'kyrs in the Icecrown area, like the ones in the Ymirheim mines (Slaves to Saronite), or maybe even the Trial of the Champion val'kyr twins scripts.

I've been using Jedoga Shadowseeker (in Ahn'Kahet) as a reference since she does some flight and landing stuff.  I assume you're saying I should just look at the DB entries for the Val'kyrs for hints? I don't think they're scripted.



> About the Boss movements you must use:
> 
> ```sql
> DELETE FROM `creature_template_movement` WHERE `CreatureId` IN (29281,30809 );
> INSERT INTO `creature_template_movement` (`CreatureId`, `Ground`, `Flights`) values
> (29281,2,1),(30809,2,1);
> ```
> 
> Also during the transformation into a val'kyr you must add bytes1=33554432 to the boss to get the correct movements (wings and visual flying )

Got those added.  Where did you pull that bytes1 value? I assume it's a bitmask, but is there any documentation as to what the other bitflags do? Or is this just a "toggle bits in the game until you get the effect you want" kind of thing?

Thanks for your help, everyone.  All aboard!
```
   _____________
 _/_|[][][][][] | - -
(  Struggle Bus | - -
=--OO-------OO--=dwb
```

**Comment by Killyana** [comment]
Val'kyrs (and some dragons) use this flag to activate the flying animation.
https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130001/creature+addon#creature_addon-bytes1

**Comment by illfated** [comment]
> Are you sure you want to learn "TrinityCore Scripting" as your second skill? Allows for a maximum skill of 75...
> 
Thank you for that in-game reference joke. Just what I needed to get a smile on my face after work. 😄

> I've been using Jedoga Shadowseeker (in Ahn'Kahet) as a reference since she does some flight and landing stuff. I assume you're saying I should just look at the DB entries for the Val'kyrs for hints? I don't think they're scripted.
> 
Fair point. from my viewpoint, it looks like you are fairly well prepared to continue.

> Thanks for your help, everyone. All aboard!
> 
> ```
>    _____________
>  _/_|[][][][][] | - -
> (  Struggle Bus | - -
> =--OO-------OO--=dwb
> ```

Hehe, thanks. Been there, done that. Didn't buy the T-shirt, though. 😉

**Comment by wotlk-enthusiast** [comment]
@Killyana 
> Val'kyrs (and some dragons) use this flag to activate the flying animation.
> https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130001/creature+addon#creature_addon-bytes1

How do I set this in the script? Looking around, I see that value (0x02000000) defined
https://github.com/TrinityCore/TrinityCore/blob/48c5c0d7a2424df73343df127d415ee18eb84296/src/server/game/Entities/Unit/UnitDefines.h#L242
but it says that only players should have that flag.  Also, when looking at other places that flag has been set, I see something like this:
https://github.com/TrinityCore/TrinityCore/blob/48c5c0d7a2424df73343df127d415ee18eb84296/src/server/scripts/Northrend/FrozenHalls/PitOfSaron/boss_krickandick.cpp#L444-L445

Am I on the wrong path?

**Comment by illfated** [comment]
I thought `SetDisableGravity()` and `SetCanFly()` would be enough to handle NPC flight, but I may have been mistaken.

**Comment by wotlk-enthusiast** [comment]
> I thought `SetDisableGravity()` and `SetCanFly()` would be enough to handle NPC flight, but I may have been mistaken.

Tried both, and they certainly make her fly around, but she hovers over the ground and walks like a priest's levitate, instead of flying around with her wings.

**Comment by illfated** [comment]
> > I thought `SetDisableGravity()` and `SetCanFly()` would be enough to handle NPC flight, but I may have been mistaken.
> 
> Tried both, and they certainly make her fly around, but she hovers over the ground and walks like a priest's levitate, instead of flying around with her wings.

Oh. Right. I forgot that there might be a different movement to get her to flap her wings. What about Olrun the Battlecaller (29047) in Scarlet Enclave, is there any useful movement inform there?

**Comment by TranceAccount** [comment]
> 
> 
> > I thought `SetDisableGravity()` and `SetCanFly()` would be enough to handle NPC flight, but I may have been mistaken.
> 
> Tried both, and they certainly make her fly around, but she hovers over the ground and walks like a priest's levitate, instead of flying around with her wings.

Have you tried using 'hover mode 2' mentioned in https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130001/creature+addon#creature_addon-bytes1 ?

**Comment by jackpoz** [comment]
what's the status of this PR ?

**Comment by illfated** [comment]
As long as the OP is not M.I.A., I guess we could ask for the file conflict to be resolved.
@wotlk-enthusiast : As above, do you have any plans to continue this?

**Comment by jackpoz** [comment]
@Carbenium could you take a look at this PR and see what's left to do before merging it ?

File: src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp [hunk 3]
```diff
@@ -303,8 +305,10 @@ class boss_svala : public CreatureScript
                             }
                             me->RemoveAllAuras();
                             me->UpdateEntry(NPC_SVALA_SORROWGRAVE);
+                            me->SetFullHealth();
                             me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
-                            events.ScheduleEvent(EVENT_INTRO_SVALA_TALK_1, 6s, 0, INTRO);
+                            me->SetByteFlag(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, UNIT_BYTE1_FLAG_ALWAYS_STAND | UNIT_BYTE1_FLAG_HOVER);
```
Comment (Warpten) [inline src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_svala.cpp:310]: No. This is `SetAnimTier(AnimTier::Fly)` at worst, either `SetDisableGravity` or `SetHover` at best.

**Comment by nerc0487** [comment]
What's the current status of this issue?

**Comment by nerc0487** [comment]
Update: had the changes done locally to check the status of the issue.

Hover is wonky, half her body is underground when in hover state (aka when she is moving):
https://imgur.com/a/SMfY2Zz
https://imgur.com/a/QCaqTIR

When she is attacking, she stands normally on the ground.

If the fight fails and you try to re-engage, she does not attack you unless you swing at her first:
https://imgur.com/a/ZA0tZIu

If the sword touches the player it does not kill him, she just inflicts around 6k damage, she does that whether or not the mobs are killed (I am unsure if the sword should kill the player if it touches them, my memory is a little flakey about Svala, been many years since I've ran wotlk utgarde)

**Comment by offl** [comment]
Did you updated your core to latest rev before applying changes from this PR?

**Comment by Carbenium** [comment]
It's on my to do list to finish this one up - I've been just a bit busy lately.

**Comment by nerc0487** [comment]
Let me know if there's anything I can help with, I'm still relatively new with fixing blizzlike issues, but would love to give a hand where possible.

Discord is Nerc#0487

**Comment by illfated** [comment]
What you can do is to create a separate, clean 3.3.5 core & DB for yourself, then apply the PR content to it, to be sure that nothing else influences any changes you make while experimenting with the PR content. You may also look into some of the latest changes and discussions, tickets and commits where @offl and others have discussed various aspects of `SetDisableGravity` / `SetHover` and related DB values.

**Comment by Aokromes** [comment]
@Carbenium  this can be merged?

**Comment by Carbenium** [comment]
Ah yes. Should be good to go. I'll rebase it tomorrow and test it once more, just to make sure nothing else broke it while it was sitting here.

**Pull request merged**